component/loki/source/file/tailer.go

```diff
@@ -61,6 +64,17 @@ func newTailer(metrics *metrics, logger log.Logger, handler loki.EntryHandler, p
 		positions.Remove(path, labels)
 	}
 
+	// If no cached position is found and the tailFromEnd option is enabled.
+	if pos == 0 && tailFromEnd {
+		pos, err = getLastLinePosition(path)
```

Comment:
Why do we only do this if no cached position is found?

I'd guess the user knows better if they want to set up this parameter. Imagine someone setting `tail_from_end = true` and then immediately stopping the Agent. They come back two days later and re-start the Agent, and the option will not be respected, but it will replay the last two days of logs.

Reply:
cc @rfratto who opened the original issue.

Reply:
That's tricky. 

If the agent crashes, you wouldn't want it to skip straight to the end on the file after restarting, and unfortunately I don't think there's any way to use just the position information to determine whether a cached position should be ignored.

I think this is safe behavior for now, but potentially something we can address in the future. 

(And to be fair, "I ran the agent for a few minutes two days ago" is likely an extreme edge case)

Reply:
I see the counterpoint, makes sense. Since we've documented this, I think we're good to go.